packages/utils/src/tabListener.ts

```diff
@@ -0,0 +1,30 @@
+const navKeys = ["Tab", "ArrowUp", "ArrowRight", "ArrowDown", "ArrowLeft"];
+let mousenavigation = false;
+
+function handleMouseDown() {
+    if (!mousenavigation) {
+        mousenavigation = true;
+        const htmlElement = document.querySelector("html");
+        htmlElement && htmlElement.setAttribute("data-mousenavigation", "true");
+        window.removeEventListener("mousedown", handleMouseDown);
+        window.addEventListener("keydown", handleKeydown);
+    }
+}
+
+function handleKeydown(event: KeyboardEvent) {
+    if (navKeys.indexOf(event.key) !== -1) {
+        if (mousenavigation) {
+            mousenavigation = false;
+            const htmlElement = document.querySelector("html");
+            htmlElement && htmlElement.removeAttribute("data-mousenavigation");
+            window.removeEventListener("keydown", handleKeydown);
+            window.addEventListener("mousedown", handleMouseDown);
+        }
+    }
+}
+
+export function initTabListener() {
```

Comment:
I wrote a test that you might consider adding:
```typescript jsx
import { fireEvent } from "@testing-library/dom";
import { initTabListener } from "./tabListener";

let mouseNavigation = "data-mousenavigation";

test("should initial mouseNavigation set to null before clicks or tabs", () => {
    initTabListener();

    const htmlElement = document.querySelector("html");

    if (!htmlElement) {
        throw 'Something went wrong  - couldn\'t find body element';
    }

    expect(htmlElement.getAttribute(mouseNavigation)).toBe(null);
});

test("should add data-mousenavigation attributen when clicking a button", () => {
    initTabListener();

    const bodyElement = document.querySelector("body");
    const htmlElement = document.querySelector("html");

    if (!bodyElement || !htmlElement) {
        throw 'Something went wrong  - couldn\'t find body element';
    }

    const buttonElement = document.createElement("button");
    bodyElement.appendChild(buttonElement);

    fireEvent(buttonElement, new MouseEvent("mousedown", { bubbles: true }));

    expect(htmlElement.getAttribute(mouseNavigation)).toBe("true");
});

test("should remove data-mousenavigation attribute when tabbing after a click", () => {
    initTabListener();

    const bodyElement = document.querySelector("body");
    const htmlElement = document.querySelector("html");

    if (!bodyElement || !htmlElement) {
        throw 'Something went wrong  - couldn\'t find body element';
    }

    const buttonElement = document.createElement("button");
    bodyElement.appendChild(buttonElement);

    fireEvent(buttonElement, new MouseEvent("mousedown", { bubbles: true }));

    expect(htmlElement.getAttribute(mouseNavigation)).toBe("true");

    fireEvent(document, new KeyboardEvent("keydown", { key: "Tab", bubbles: true }));

    expect(htmlElement.getAttribute(mouseNavigation)).toBe(null);
});

```

Reply:
Oh. And we need to install @testing-library/dom if we use this test.

Reply:
Nice tests! They are now shamelessly stolen 😅